source/grid/grid_tools_dof_handlers.cc

```diff
@@ -2261,20 +2261,20 @@ namespace GridTools
    *   matrix.point1 + offset - point2
    * is parallel to the unit vector in <direction>
    */
-  template <int spacedim>
+  template <int spacedim, typename number>
   inline bool
-  orthogonal_equality(const Point<spacedim> &    point1,
-                      const Point<spacedim> &    point2,
-                      const int                  direction,
-                      const Tensor<1, spacedim> &offset,
-                      const FullMatrix<double> & matrix)
+  orthogonal_equality(const Point<spacedim> &            point1,
+                      const Point<spacedim> &            point2,
+                      const int                          direction,
+                      const Tensor<1, spacedim, number> &offset,
+                      const FullMatrix<number> &         matrix)
```

Comment:
`point1` and `point2` are real, therefore the transformation between `point1` and `point2` has to be real.